File: iCyption2022/CftTest/June/solution.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef vector<ll> vll;

ll LCM(ll a, ll b) {
    return (a / __gcd(a, b)) * b;
}

int main() {
    int n, q; cin >> n >> q;

    vll v;
    bool exist[2501] = {false};
    for (int i = 1; i <= 50; i++) {
        for (int j = i+1; j <= 50; j++) {
            ll lcm = LCM(i, j);
            if (lcm > 100 && !exist[lcm]) {
                v.push_back(lcm);
                exist[lcm] = true;
            }
        }
    }

    ll sume[2501] = {0};
    ll sum = 0;

    for (int j = 0; j < n; j++) {
        ll num; cin >> num;
        sum += num;
        for (int i = 1; i <= 100; i++) 
            sume[i] += (num % i == 0 ? num : 0);
        
        for (ll i : v)
            sume[i] += (num % i == 0 ? num : 0);
    }

    for (int j = 0; j < q; j++) {
        int m; cin >> m;
        ll a[m];

        for (int i = 0; i < m; i++)
            cin >> a[i];

        if (m == 1)
            cout << sume[1] - sume[a[0]];
        else 
            cout << sume[1] - sume[a[0]] - sume[a[1]] + sume[LCM(a[0], a[1])];

        cout << '\n';
    }
}

File: iCyption2022/CftTest/June/solution.py
def gcd(a, b) :
    if (b == 0):
        return a
    return gcd(b, a % b)

def lcm(a, b) :
    return (a // gcd(a, b)) * b

n, q = map(int, input().split())

v = []
exist = [False for i in range(2501)]
for i in range(1, 51) :
    for j in range(i + 1, 51) :
        LCM = lcm(i, j)
        if (LCM > 100 and not exist[LCM]) :
            v.append(LCM)
            exist[LCM] = True

sume = [0 for i in range(2501)]
sum = 0

num = list(map(int, input().split()))
for j in range(n) :
    sum += num[j]
    for i in range(1, 101) :
        sume[i] += (num[j] if num[j] % i == 0 else 0)

    for i in v:
        sume[i] += (num[j] if num[j] % i == 0 else 0)
    
for j in range(q) :
    m = int(input())
    a = list(map(int, input().split()))

    if (m == 1) :
        print(sume[1] - sume[a[0]])
    else :
        print(sume[1] - sume[a[0]] - sume[a[1]] + sume[lcm(a[0], a[1])])